src/main/java/org/sonarlint/intellij/config/global/SonarQubeServer.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.config.global;
+
+import com.intellij.openapi.util.Comparing;
+import com.intellij.openapi.util.PasswordUtil;
+import com.intellij.util.xmlb.annotations.Tag;
+import com.intellij.util.xmlb.annotations.Transient;
+
+public class SonarQubeServer {
+  private String hostUrl;
+  private String token;
+
+  private String name;
+  private String login;
+  private String password;
+  private String serverId;
+  private boolean enableProxy;
+
+  public SonarQubeServer() {
+
+  }
+
+  public SonarQubeServer(SonarQubeServer toCopy) {
+    setHostUrl(toCopy.getHostUrl());
+    setServerId(toCopy.getServerId());
+    setPassword(toCopy.getPassword());
+    setToken(toCopy.getToken());
+    setLogin(toCopy.getLogin());
+    setName(toCopy.getName());
+    setEnableProxy(toCopy.enableProxy());
+  }
+
+  @Override
+  public boolean equals(Object o){
+    if (!(o instanceof SonarQubeServer)) return false;
+    SonarQubeServer other = (SonarQubeServer)o;
+
+    return Comparing.equal(getHostUrl(), other.getHostUrl()) &&
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (6) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1067)